common/Makefile.am

```diff
@@ -47,7 +47,12 @@ libcommon_la_SOURCES = \
   xrdp_rail.h \
   $(PIXMAN_SOURCES)
 
+if NEED_LIBDL
+EXTRA_LIBDL = -ldl
+endif
+
 libcommon_la_LIBADD = \
+  $(EXTRA_LIBDL) \
```

Comment:
EXTRA_LIBDL is a poor name. I would prefer OPENSSL_LIBS, which would get `-ldl` added to it conditionally. A bigger issue is that we should support pkg-config for OpenSSL and take the settings from there.